lib/ecto/query/planner.ex

```diff
@@ -1309,7 +1309,28 @@ defmodule Ecto.Query.Planner do
     if Ecto.Type.base?(type) do
       {tagged, acc}
     else
-      {dump_param(kind, query, expr, v, type, adapter), acc}
+      type = field_type!(kind, query, expr, type)
```

Comment:
Sorry if this is noise but I remember playing around with this and you might have to call `Ecto.Type.type` on this to get `:uuid`? Otherwise I think it will be `Ecto.UUID`

Reply:
It is all good as is. `field_type` gets the type from a field, which may still be `Ecto.UUID`. But the Ecto.Type calls always reserve the type properly. :)

Reply:
oh you are already considering that below. ignore me!